CMakeLists.txt

```diff
@@ -1,6 +1,32 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 2.8.7) #Currently targeting to TRAVIS CI cmake version from 2011
+foreach(p
+    CMP0025 # CMake 3.0 Compiler id for Apple Clang is now ``AppleClang``.
+    CMP0042 # CMake 3.0 ``MACOSX_RPATH`` is enabled by default.
+    CMP0046 # CMake 3.0 Error on non-existent dependency in add_dependencies.
+    CMP0054 # CMake 3.1 Only interpret ``if()`` arguments as variables or keywords when unquoted.
+    CMP0056 # CMake 3.2 Honor link flags in ``try_compile()`` source-file signature.
+    CMP0058 # CMake 3.3 Ninja requires custom command byproducts to be explicit.
+    )
+  if(POLICY ${p})
+    cmake_policy(SET ${p} NEW)
+  endif()
+endforeach()
 project(GADGETRON)
 
+# --- Provide good defaults for searching for packages (i.e. ismrmrd)
+if(CMAKE_INSTALL_PREFIX)
+  list(APPEND CMAKE_INSTALL_PREFIX "/usr/local")
```

Comment:
GOOD CATCH!

 

You are exactly correct.  I forgot that CMAKE_INSTALL_PREFIX is the variable name used to install packages.  I can not fix that until later tonight.

 

 

 

Hans

 

 

-- 

 

 

From: Michael Hansen <notifications@github.com>
Reply-To: gadgetron/gadgetron <reply@reply.github.com>
Date: Thursday, June 30, 2016 at 12:48 PM
To: gadgetron/gadgetron <gadgetron@noreply.github.com>
Cc: Hans Johnson <hans.j.johnson@gmail.com>, Author <author@noreply.github.com>
Subject: Re: [gadgetron/gadgetron] Update to new ismrmrd cmake conventions (#402)

 

In CMakeLists.txt:
>  project(GADGETRON)
>  
> +# --- Provide good defaults for searching for packages (i.e. ismrmrd)
> +if(CMAKE_INSTALL_PREFIX)
> +  list(APPEND CMAKE_INSTALL_PREFIX "/usr/local")
I think this is wrong. It is not really needed since it is part of the search path and it ends up modifying where things are installed and not where it searches, right?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.